src/source/src/parser/maxwell/json.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::BTreeMap;
+use std::fmt::Debug;
+
+use risingwave_common::error::ErrorCode::ProtocolError;
+use risingwave_common::error::{Result, RwError};
+use serde_derive::{Deserialize, Serialize};
+use serde_json::Value;
+
+use crate::parser::common::json_parse_value;
+use crate::{SourceParser, SourceStreamChunkRowWriter, WriteGuard};
+
+const MAXWELL_INSERT_OP: &str = "insert";
+const MAXWELL_UPDATE_OP: &str = "update";
+const MAXWELL_DELETE_OP: &str = "delete";
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MaxwellEvent {
+    pub data: Option<BTreeMap<String, Value>>,
+    pub old: Option<BTreeMap<String, Value>>,
+    #[serde(rename = "type")]
+    pub op: String,
+    #[serde(rename = "ts")]
+    pub ts_ms: i64,
+}
+
+#[derive(Debug)]
+pub struct MaxwellParser;
+
+impl SourceParser for MaxwellParser {
+    fn parse(&self, payload: &[u8], writer: SourceStreamChunkRowWriter<'_>) -> Result<WriteGuard> {
+        let event: MaxwellEvent = serde_json::from_slice(payload)
+            .map_err(|e| RwError::from(ProtocolError(e.to_string())))?;
+
+        match event.op.as_str() {
+            MAXWELL_INSERT_OP => {
+                let after = event.data.as_ref().ok_or_else(|| {
+                    RwError::from(ProtocolError(
+                        "data is missing for creating event".to_string(),
+                    ))
+                })?;
+                writer.insert(|column| {
+                    json_parse_value(&column.data_type, after.get(&column.name)).map_err(Into::into)
```

Comment:
Why maxwell and debezium parser don't use `simd_json_parse_value`.

Reply:
Will try with simd-json